sdk/core/Azure.Core.TestFramework/src/TestProxy.cs

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+using NUnit.Framework;
+
+namespace Azure.Core.TestFramework
+{
+    /// <summary>
+    /// Encapsulates a process running an instance of the Test Proxy as well as providing access to the Test Proxy administration actions
+    /// via the TestProxyRestClient.
+    /// <seealso href="https://github.com/Azure/azure-sdk-tools/tree/main/tools/test-proxy"/>
+    /// </summary>
+    public class TestProxy
+    {
+        private static readonly string s_dotNetExe;
+
+        public const string DevCertIssuer = "CN=localhost";
+
+        public const string IpAddress = "127.0.0.1";
+
+        public int? ProxyPortHttp => _proxyPortHttp;
+
+        public int? ProxyPortHttps => _proxyPortHttps;
+
+        private readonly int? _proxyPortHttp;
+        private readonly int? _proxyPortHttps;
+        private readonly Process _testProxyProcess;
+        internal TestProxyRestClient Client { get; }
+        private readonly StringBuilder _errorBuffer = new();
+
+        static TestProxy()
+        {
+            string installDir = Environment.GetEnvironmentVariable("DOTNET_INSTALL_DIR");
+            var dotNetExeName = "dotnet" + (Path.DirectorySeparatorChar == '/' ? "" : ".exe");
+            if (!HasDotNetExe(installDir))
+            {
+                installDir = Environment.GetEnvironmentVariable("PATH")?.Split(Path.PathSeparator).FirstOrDefault(HasDotNetExe);
+            }
+
+            if (installDir == null)
+            {
+                throw new InvalidOperationException("DOTNET install directory was not found");
+            }
+
+            s_dotNetExe = Path.Combine(installDir, dotNetExeName);
+
+            bool HasDotNetExe(string dotnetDir) => dotnetDir != null && File.Exists(Path.Combine(dotnetDir, dotNetExeName));
+        }
+
+        private TestProxy(string proxyPath)
+        {
+            ProcessStartInfo testProxyProcessInfo = new ProcessStartInfo(
+                s_dotNetExe,
+                proxyPath)
+            {
+                UseShellExecute = false,
+                RedirectStandardOutput = true,
+                RedirectStandardError = true,
+                EnvironmentVariables =
+                {
+                    ["ASPNETCORE_URLS"] = $"http://{IpAddress}:0;https://{IpAddress}:0",
+                    ["Logging__LogLevel__Microsoft"] = "Information"
+                }
+            };
+
+            if (!TestEnvironment.GlobalIsRunningInCI)
+            {
+                ImportDevCertIfNeeded();
+            }
+
+            _testProxyProcess = Process.Start(testProxyProcessInfo);
```

Comment:
You should `using` this as it's `IDisposable`.

Reply:
Rather, since it's a field, should the whole proxy be `IDisposable` or, in this case, is the lifetime management not a concern. Without having reviewed all the code just yet, is there only one proxy per test run, thus lifetime is not really a problem?

Reply:
The proxy is per test class - it's lifetime is managed via GlobalSetup and GlobalTeardown attributes in RecordedTestBase.